hw/bitstream/BUILD

```diff
@@ -57,17 +59,44 @@ filegroup(
     name = "mask_rom",
     srcs = select({
         "bitstream_skip": ["skip.bit"],
-        "bitstream_vivado": ["//hw/bitstream/vivado:fpga_cw310_mask_rom"],
+        "bitstream_vivado": [":fpga_cw310_mask_rom"],
         "bitstream_gcp": ["@bitstreams//:bitstream_mask_rom"],
         "//conditions:default": ["@bitstreams//:bitstream_mask_rom"],
     }),
 )
 
-# TODO(lowRISC/opentitan#13603): Use `select` once we're uploading these
-# artifacts to GCP.
 filegroup(
-    name = "mask_rom_otp_dev",
-    srcs = [
-        "//hw/bitstream/vivado:fpga_cw310_mask_rom_otp_dev",
-    ],
+    name = "rom_mmi",
+    srcs = select({
+        "bitstream_skip": ["skip.bit"],
+        "bitstream_vivado": ["//hw/bitstream/vivado:rom_mmi"],
+        "bitstream_gcp": ["@bitstreams//:rom_mmi"],
+        "//conditions:default": ["@bitstreams//:rom_mmi"],
+    }),
+)
+
+filegroup(
+    name = "otp_mmi",
+    srcs = select({
+        "bitstream_skip": ["skip.bit"],
+        "bitstream_vivado": ["//hw/bitstream/vivado:otp_mmi"],
+        "bitstream_gcp": ["@bitstreams//:otp_mmi"],
+        "//conditions:default": ["@bitstreams//:otp_mmi"],
+    }),
+)
+
+bitstream_splice(
+    name = "fpga_cw310_mask_rom",
+    src = ":test_rom",
+    data = "//sw/device/silicon_creator/mask_rom:mask_rom_fpga_cw310_scr_vmem",
+    meminfo = ":rom_mmi",
+    tags = ["manual"],
+)
```

Comment:
I'm not sure we have an explicit use case for this rule. A case an be easily imagined:

I add something to the ROM and want to quickly test it by splicing the ROM built in my workspace into an already prepared bitstream.

Since all of the existing test rules depend on either `:test_rom` or `:mask_rom`, you'll need some way of selecting this rule via the existing `:{test,mask}_rom` filegroups. To understand how to select such a thing and how to write the `src` for this rule, you have to basically double the bitstream selection criteria and then eliminate the combinations that don't make sense:

```
--define bitstream=...
    <undefined> - same as gcp (ie: no --define bitstream=...).
    skip - no bitstream reloading.
    vivado - build the bitstream with vivado (which would use the bitstream in the workspace).
    gcp - use the bitstreams-cache supplied bitstream.
    splice_skip - This option doesn't make sense since it won't do anything..
    splice_gcp - splice the ROM in the workspace into the bitstream from GCP.
    splice_vivado - same as vivado (since the vivado targets have a dependency on the ROM).
```

I think this means one can then re-run tests against a changed {Test,Mask} ROM with one of `vivado` or `splice_gcp`.  `splice_skip` and `splice_vivado` aren't needed (and once we get to building the bitstream with an empty rom and having splices for both test/mask rom, then `--define bitstream=vivado` will only have to build the bitstream once and then just do splices for the ROMs as you iterate).

Reply:
Great point. I'm addressing this in the next patch and reworking the documentation a little.